io-engine/src/bdev/nexus/nexus_bdev_rebuild.rs

```diff
@@ -417,6 +430,16 @@ impl<'n> Nexus<'n> {
 
         match job_state {
             RebuildState::Completed => {
+                self.event(
+                    EventAction::RebuildEnd,
+                    rebuild_event_meta(
+                        RebuildStatus::Completed,
+                        job.src_uri(),
+                        child_uri,
+                        None,
+                    ),
```

Comment:
Can't we generate this through job?
Example job.event_meta() ?

Reply:
Yes we can. Will add it.

Reply:
I can add RebuildStatus directly from the job.state(). Should I add it as string instead of using enum in protobuf?

Reply:
I mean generate the entire rebuild_event meta from the job itself.
```rust
self.event(EventAction::RebuildBegin, job.event_meta())
```

Reply:
Yes. 
impl EventMetaGen for RebuildJob {
    fn meta(&self) -> EventMeta {
        let event_source =
        EventSource::new(MayastorEnvironment::global_or_default().node_name)
            .add_rebuild_data(&self.state().to_string(), self.src_uri(), self.dst_uri(), &self.error_desc());
        EventMeta::from_source(event_source)
    }
 }
 
 This is how I am trying to generate the event_meta. 
 
 Here I can get the status from status() itself. So just wanted to check if I can use string instead of adding another enum in protobuf.
 
 Also wanted to check if error should be optional because error_desc() directly gives the empty string if there is no error.

Reply:
Added the changes.